gravitee-apim-repository/gravitee-apim-repository-jdbc/src/main/java/io/gravitee/repository/jdbc/management/JdbcPortalNavigationItemRepository.java

```diff
@@ -64,26 +64,49 @@ public List<PortalNavigationItem> findAllByOrganizationIdAndEnvironmentId(String
     }
 
     @Override
-    public List<PortalNavigationItem> findAllByAreaAndOrganizationIdAndEnvironmentId(
-        PortalNavigationItem.Area area,
-        String organizationId,
-        String environmentId
-    ) throws TechnicalException {
-        log.debug(
-            "JdbcPortalNavigationItemRepository.findAllByAreaAndOrganizationIdAndEnvironmentId({}, {}, {})",
-            area,
-            organizationId,
-            environmentId
-        );
+    public List<PortalNavigationItem> findAllByParentIdAndEnvironmentId(String parentId, String environmentId) throws TechnicalException {
+        log.debug("JdbcPortalNavigationItemRepository.findAllByParentIdAndEnvironmentId({}, {})", parentId, environmentId);
         try {
-            final String sql = getOrm().getSelectAllSql() + " where area = ? and organization_id = ? and environment_id = ?";
-            return jdbcTemplate.query(sql, getOrm().getRowMapper(), area.name(), organizationId, environmentId);
+            final String sql =
+                "select id, organization_id, environment_id, title, type, area, parent_id, \"order\", configuration from " +
+                this.tableName +
+                " where parent_id = ? and environment_id = ?";
+            return jdbcTemplate.query(sql, getOrm().getRowMapper(), parentId, environmentId);
+        } catch (Exception ex) {
+            log.error("Failed to find portal navigation items by parentId", ex);
+            throw new TechnicalException("Failed to find portal navigation items by parentId", ex);
+        }
+    }
+
+    @Override
+    public List<PortalNavigationItem> findAllByAreaAndEnvironmentId(PortalNavigationItem.Area area, String environmentId)
+        throws TechnicalException {
+        log.debug("JdbcPortalNavigationItemRepository.findAllByAreaAndEnvironmentId({}, {})", area, environmentId);
+        try {
+            final String sql = getOrm().getSelectAllSql() + " where area = ? and environment_id = ?";
+            return jdbcTemplate.query(sql, getOrm().getRowMapper(), area.name(), environmentId);
         } catch (Exception ex) {
             log.error("Failed to find portal navigation items by area", ex);
             throw new TechnicalException("Failed to find portal navigation items by area", ex);
         }
     }
 
+    @Override
+    public List<PortalNavigationItem> findAllByAreaAndEnvironmentIdAndParentIdIsNull(PortalNavigationItem.Area area, String environmentId)
```

Comment:
Could you merge this method with `findAllByParentIdAndEnvironmentId` ?
It could give something like this:

```suggestion
    @Override
    public List<PortalNavigationItem> findAllByAreaAndEnvironmentIdAndParentId(PortalNavigationItem.Area area, String environmentId, String parentId) throws TechnicalException {
        log.debug("JdbcPortalNavigationItemRepository.findAllByAreaAndEnvironmentIdAndParentId({}, {}, {})", area, environmentId, parentId);
        try {
            String sql = getOrm().getSelectAllSql() + " where area = ? and environment_id = ?";
            if (parentId != null) {
                sql += " and parent_id = ?";
            } else {
                sql += " and parent_id is null";
            }
            return jdbcTemplate.query(sql, getOrm().getRowMapper(), area.name(), environmentId, parentId);
        } catch (Exception ex) {
            throw new TechnicalException("Failed to find portal navigation items by area, environment and parentId", ex);
        }
    }
```

Reply:
As discussed, let's keep them separate